pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -317,6 +320,62 @@ func (vs *vSphereVMProvider) GetVirtualMachineHardwareVersion(
 	return vimtypes.ParseHardwareVersion(o.Config.Version)
 }
 
+func (vs *vSphereVMProvider) vmCreatePathName(
+	vmCtx pkgctx.VirtualMachineContext,
+	vcClient *vcclient.Client,
+	createArgs *VMCreateArgs) error {
+
+	if createArgs.StorageProfileID == "" {
+		return nil
+	}
+	if createArgs.ConfigSpec.Files == nil {
+		createArgs.ConfigSpec.Files = &vimtypes.VirtualMachineFileInfo{}
+	}
+	if createArgs.ConfigSpec.Files.VmPathName != "" {
+		return nil
+	}
+
+	// TODO: we should only do this when deploying ISO library items
```

Comment:
There are some other tasks also depending on deploying ISO library items (e.g., skipping default bootstrap). Might be worth filing a JIRA/issue to track and address them together.